File: C#-CarRacingProject/Repository/DB/InscriereDbRepo.cs
using log4net;
using ProiectCSharpVersion.Domain;
using ProiectCSharpVersion.Repository.Interfaces;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WindowsFormApp1.Repository;

namespace ProiectCSharpVersion.Repository.DB
{
    class InscriereDbRepo : IInscriereRepository
    {
        private static readonly ILog log = LogManager.GetLogger("CursaDbRepo");
        public static ParticipantDbRepo pdr;
        private static CursaDbRepo cdr;
        public InscriereDbRepo(ParticipantDbRepo pdrr, CursaDbRepo cdrr)
        {
            log.Info("Creating ParticipantDbRepo");
            pdr = pdrr;
            cdr = cdrr;
        }
        public Inscriere Delete(int id)
        {
            log.InfoFormat("Entering Update");
            IDbConnection con = DBUtils.getConnection();
            Inscriere ins = FindOne(id);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "delete from Inscriere where iid=@iid";

                IDbDataParameter paramIid = comm.CreateParameter();
                paramIid.ParameterName = "@iid";
                paramIid.Value = id;
                comm.Parameters.Add(paramIid);

                var result = comm.ExecuteNonQuery();
                if (result == 0)
                {
                    log.InfoFormat("update exited with value {0}", null);
                    return null;
                }
                else
                {
                    log.InfoFormat("update exited with value {0}", result);
                    return ins;
                }
            }
        }

        public IEnumerable<Inscriere> FindAll()
        {
            log.InfoFormat("Entering findAll");
            IDbConnection con = DBUtils.getConnection();
            IList<Inscriere> inscrieri = new List<Inscriere>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Inscriere";
                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int iid = dataR.GetInt32(0);
                        int pid = dataR.GetInt32(1);
                        int cid = dataR.GetInt32(2);
                        Inscriere i = new Inscriere(pdr.FindOne(pid), cdr.FindOne(cid));
                        i.Id = iid;
                        inscrieri.Add(i);
                    }
                }
            }
            log.InfoFormat("Exiting findAll");
            return inscrieri;
        }

        public Inscriere FindOne(int id)
        {
            log.InfoFormat("Entering findOne");
            IDbConnection con = DBUtils.getConnection();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select iid,pid,cid from Inscriere where iid=@iid";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@iid";
                paramId.Value = id;
                comm.Parameters.Add(paramId);
                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        int iid = dataR.GetInt32(0);
                        int pid = dataR.GetInt32(1);
                        int cid = dataR.GetInt32(2);
                        Inscriere i = new Inscriere(pdr.FindOne(pid), cdr.FindOne(cid));
                        i.Id = iid;
                        log.InfoFormat("Exiting findOne with value {0}", i);
                        return i;
                    }
                }
            }
            log.InfoFormat("Exiting findOne with value {0}", null);
            return null;
        }

        public IEnumerable<Inscriere> GetInscrieriByIdCursa(int idCursa)
        {
            log.InfoFormat("Entering GetInscrieriByIdCursa");
            IDbConnection con = DBUtils.getConnection();
            IList<Inscriere> inscrieri = new List<Inscriere>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select iid,pid,cid from Inscriere where cid=@cid";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@cid";
                paramId.Value = idCursa;
                comm.Parameters.Add(paramId);
                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int iid = dataR.GetInt32(0);
                        int pid = dataR.GetInt32(1);
                        int cid = dataR.GetInt32(2);
                        Inscriere i = new Inscriere(pdr.FindOne(pid), cdr.FindOne(cid));
                        i.Id = iid;
                        inscrieri.Add(i);
                    }
                }
            }
            log.InfoFormat("Exiting GetInscrieriByIdCursa");
            return inscrieri;
        }

        public IEnumerable<Inscriere> GetInscrieriByIdParticipant(int idParticipant)
        {
            log.InfoFormat("Entering GetInscrieriByIdParticipant");
            IDbConnection con = DBUtils.getConnection();
            IList<Inscriere> inscrieri = new List<Inscriere>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select iid,pid,cid from Inscriere where pid=@pid";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@pid";
                paramId.Value = idParticipant;
                comm.Parameters.Add(paramId);
                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int iid = dataR.GetInt32(0);
                        int pid = dataR.GetInt32(1);
                        int cid = dataR.GetInt32(2);
                        Inscriere i = new Inscriere(pdr.FindOne(pid), cdr.FindOne(cid));
                        i.Id = iid;
                        inscrieri.Add(i);
                    }
                }
            }
            log.InfoFormat("Exiting GetInscrieriByIdParticipant");
            return inscrieri;
        }

        public Inscriere Save(Inscriere entity)
        {
            log.InfoFormat("Entering Save");
            IDbConnection con = DBUtils.getConnection();

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "Insert into Inscriere(pid,cid) values (@pid,@cid)";

                IDbDataParameter paramPid = comm.CreateParameter();
                paramPid.ParameterName = "@pid";
                paramPid.Value = entity.Participant.Id;
                comm.Parameters.Add(paramPid);

                IDbDataParameter paramCid = comm.CreateParameter();
                paramCid.ParameterName = "@cid";
                paramCid.Value = entity.Cursa.Id;
                comm.Parameters.Add(paramCid);

                var result = comm.ExecuteNonQuery();
                if (result == 0)
                {
                    log.InfoFormat("save exited with value {0}", null);
                    return null;
                }
                else
                {
                    log.InfoFormat("save exited with value {0}", result);
                    return entity;
                }
            }
        }

        public Inscriere Update(Inscriere entity)
        {
            log.InfoFormat("Entering Update");
            IDbConnection con = DBUtils.getConnection();

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "update Inscriere set pid=@pid, cid=@cid where iid=@iid";

                IDbDataParameter paramPid = comm.CreateParameter();
                paramPid.ParameterName = "@pid";
                paramPid.Value = entity.Participant.Id;
                comm.Parameters.Add(paramPid);

                IDbDataParameter paramCid = comm.CreateParameter();
                paramCid.ParameterName = "@cid";
                paramCid.Value = entity.Cursa.Id;
                comm.Parameters.Add(paramCid);

                IDbDataParameter paramIid = comm.CreateParameter();
                paramIid.ParameterName = "@iid";
                paramIid.Value = entity.Id;
                comm.Parameters.Add(paramIid);

                var result = comm.ExecuteNonQuery();
                if (result == 0)
                {
                    log.InfoFormat("update exited with value {0}", null);
                    return null;
                }
                else
                {
                    log.InfoFormat("update exited with value {0}", result);
                    return entity;
                }
            }
        }
    }
}

File: CarRacing_RestServices/CursePersistence/src/main/java/curse/repository/database/InscriereDbRepo.java
package curse.repository.database;

import curse.model.Inscriere;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import curse.repository.interfaces.IInscriereRepository;

import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.List;
import java.sql.Connection;
import java.util.ArrayList;
import java.util.Properties;

public class InscriereDbRepo implements IInscriereRepository {
    private final Jdbc dbUtils;
    private static final Logger logger= LogManager.getLogger();
    private final ParticipantDbRepo participantDbRepo;
    private final CursaDbRepo cursaDbRepo;
    public InscriereDbRepo(Properties properties, ParticipantDbRepo participantDbRepo, CursaDbRepo cursaDbRepo){
        logger.info("Initializare InscriereDbRepo cu proprietatile {}",properties);
        dbUtils=new Jdbc(properties);
        this.participantDbRepo=participantDbRepo;
        this.cursaDbRepo=cursaDbRepo;
    }

    @Override
    public Iterable<Inscriere> findInscrieribyIdParticipant(Integer idParticipant) {
        logger.traceEntry();
        Connection connection=dbUtils.getConnection();
        List<Inscriere> inscrieri=new ArrayList<>();
        try(PreparedStatement prepStat=connection.prepareStatement("SELECT * FROM Inscriere WHERE pid=?")){
            prepStat.setInt(1,idParticipant);
            try(ResultSet result=prepStat.executeQuery()){
                while(result.next()){
                    int iid=result.getInt("iid");
                    int pid=result.getInt("pid");
                    int cid=result.getInt("cid");
                    Inscriere i= new Inscriere(participantDbRepo.findOne(pid),cursaDbRepo.findOne(cid));
                    i.setId(iid);
                    inscrieri.add(i);
                }
            }
        }catch(SQLException e){
            logger.error(e);
            System.err.println("DB Error"+e);
        }
        return inscrieri;
    }

    @Override
    public Iterable<Inscriere> findInscrieribyIdCursa(Integer idCursa) {
        logger.traceEntry();
        Connection connection=dbUtils.getConnection();
        List<Inscriere> inscrieri=new ArrayList<>();
        try(PreparedStatement prepStat=connection.prepareStatement("SELECT * FROM Inscriere WHERE cid=?")){
            prepStat.setInt(1,idCursa);
            try(ResultSet result=prepStat.executeQuery()){
                while(result.next()){
                    int iid=result.getInt("iid");
                    int pid=result.getInt("pid");
                    int cid=result.getInt("cid");
                    Inscriere i= new Inscriere(participantDbRepo.findOne(pid),cursaDbRepo.findOne(cid));
                    i.setId(iid);
                    inscrieri.add(i);
                }
            }
        }catch(SQLException e){
            logger.error(e);
            System.err.println("DB Error"+e);
        }
        return inscrieri;
    }

    @Override
    public Inscriere findOne(Integer integer) {
        logger.traceEntry();
        Connection connection=dbUtils.getConnection();
        Inscriere i =new Inscriere(null,null);
        try(PreparedStatement prepStat=connection.prepareStatement("SELECT * FROM Inscriere WHERE iid=?")){
            prepStat.setInt(1,integer);
            try(ResultSet result=prepStat.executeQuery()){
                int iid=result.getInt("iid");
                int pid=result.getInt("pid");
                int cid=result.getInt("cid");
                i.setParticipant(participantDbRepo.findOne(pid));
                i.setCursa(cursaDbRepo.findOne(cid));
                i.setId(iid);
            }
        }catch(SQLException e){
            logger.error(e);
        }
        if(i.getCursa()==null)
            return null;
        else
            return i;
    }

    @Override
    public Iterable<Inscriere> findAll() {
        logger.traceEntry();
        Connection connection=dbUtils.getConnection();
        List<Inscriere> inscrieri=new ArrayList<>();
        try(PreparedStatement prepStat=connection.prepareStatement("SELECT * FROM Inscriere")){
            try(ResultSet result=prepStat.executeQuery()){
                while(result.next()){
                    int iid=result.getInt("iid");
                    int pid=result.getInt("pid");
                    int cid=result.getInt("cid");
                    Inscriere i= new Inscriere(participantDbRepo.findOne(pid),cursaDbRepo.findOne(cid));
                    i.setId(iid);
                    inscrieri.add(i);
                }
            }
        }catch(SQLException e){
            logger.error(e);
            System.err.println("DB Error"+e);
        }
        return inscrieri;
    }

    @Override
    public Inscriere save(Inscriere entity) {
        logger.traceEntry();
        Connection connection=dbUtils.getConnection();
        Inscriere i =findOne(entity.getId());
        if(i!=null)
            return null;
        try(PreparedStatement prepStat=connection.prepareStatement("INSERT INTO Inscriere(pid,cid) VALUES (?,?)")){
            prepStat.setInt(1,entity.getParticipant().getId());
            prepStat.setInt(2,entity.getCursa().getId());
            prepStat.executeUpdate();
        }catch(SQLException e){
            logger.error(e);
            System.err.println("DB Error"+e);
        }
        logger.traceExit();
        return entity;
    }

    @Override
    public Inscriere delete(Integer integer) {
        logger.traceEntry();
        Connection connection=dbUtils.getConnection();
        Inscriere i =findOne(integer);
        if(i==null)
            return null;
        try(PreparedStatement prepStat=connection.prepareStatement("DELETE FROM Inscriere WHERE iid=?")){
            prepStat.setInt(1,integer);
            prepStat.executeUpdate();
        }catch(SQLException e){
            logger.error(e);
            System.err.println("DB Error"+e);
        }
        logger.traceExit();
        return i;
    }

    @Override
    public Inscriere update(Inscriere entity) {
        logger.traceEntry();
        Connection connection=dbUtils.getConnection();
        Inscriere i =findOne(entity.getId());
        if(i==null)
            return null;
        try(PreparedStatement prepStat=connection.prepareStatement("UPDATE Inscriere SET pid=?, cid=? WHERE iid=?")){
            prepStat.setInt(1,entity.getParticipant().getId());
            prepStat.setInt(2,entity.getCursa().getId());
            prepStat.setInt(3,entity.getId());
            prepStat.executeUpdate();
        }catch(SQLException e){
            logger.error(e);
            System.err.println("DB Error"+e);
        }
        logger.traceExit();
        return entity;
    }
}
